File: solutions/_387_first_unique_character_in_a_string/solution.py
from collections import Counter


class Solution:
    def firstUniqChar(self, s: str) -> int:
        ctr = Counter(s)
        for i, char in enumerate(s):
            if ctr.get(char) == 1:
                return i
        return -1


# https://leetcode.com/problems/first-unique-character-in-a-string/

File: solutions/_387_first_unique_character_in_a_string/solution.cpp
#include <string>
using namespace std;

class Solution
{
public:
    int firstUniqChar(string s)
    {
        int freq[26] = {};
        for (char c : s)
        {
            freq[c - 'a']++;
        }
        for (int i = 0; i < s.size(); i++)
        {
            if (freq[s[i] - 'a'] == 1)
            {
                return i;
            }
        }
        return -1;
    }
};

/*
https://leetcode.com/problems/first-unique-character-in-a-string/
*/